src/utils/form/getErrorMessage.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+
+function getErrorMessage(meta: FinalFormMeta, validateType: ?JInputValidateType): ?string {
+  const err = meta.error || meta.submitError
+
+  const error = (typeof err === 'string')
+    ? err
+    : err instanceof Error && err.message
+      ? err.message
+      : ''
+
+  if (validateType === 'dirtySinceLastSubmit' && !meta[validateType]) {
+    return error
+  } else if ((validateType === 'touched' || validateType === 'visited') && meta[validateType]) {
```

Comment:
let's create specific vars, like isDirty, isTouched, isVisited, isExist(for meta[validateType])

currently really hard to read